nat-lab/tests/test_mesh_firewall.py

```diff
@@ -525,3 +525,283 @@ async def on_stdout(stdout: str) -> None:
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+            )
+        )
+        (connection_beta, beta_conn_tracker) = await exit_stack.enter_async_context(
+            new_connection_with_conn_tracker(
+                ConnectionTag.DOCKER_CONE_CLIENT_2,
+                generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+            )
+        )
+
+        client_alpha = await exit_stack.enter_async_context(
+            telio.Client(
+                connection_alpha, alpha, telio.AdapterType.BoringTun
+            ).run_meshnet(api.get_meshmap(alpha.id))
+        )
+
+        client_beta = await exit_stack.enter_async_context(
+            telio.Client(
+                connection_beta, beta, telio.AdapterType.BoringTun
+            ).run_meshnet(api.get_meshmap(beta.id))
+        )
+
+        await testing.wait_long(
+            asyncio.gather(
+                client_alpha.wait_for_any_derp_state([telio.State.Connected]),
+                client_beta.wait_for_any_derp_state([telio.State.Connected]),
+            )
+        )
+
+        await testing.wait_long(
+            asyncio.gather(
+                alpha_conn_tracker.wait_for_event("derp_1"),
+                beta_conn_tracker.wait_for_event("derp_1"),
+            )
+        )
+
+        await testing.wait_long(
+            asyncio.gather(
+                client_alpha.handshake(beta.public_key),
+                client_beta.handshake(alpha.public_key),
+            )
+        )
+
+        output_notifier = OutputNotifier()
+
+        async def on_stdout(stdout: str) -> None:
+            for line in stdout.splitlines():
+                output_notifier.handle_output(line)
+
+        async def conntrack_on_stdout(stdout: str) -> None:
+            for line in stdout.splitlines():
+                if f"src={CLIENT_BETA_IP} dst={CLIENT_ALPHA_IP}" in line:
+                    output_notifier.handle_output(line)
+
+        listening_start_event = asyncio.Event()
+        sender_start_event = asyncio.Event()
+        connected_event = asyncio.Event()
+        last_ack_event = asyncio.Event()
+        time_wait_event = asyncio.Event()
+
+        output_notifier.notify_output(
+            f"listening on [{CLIENT_ALPHA_IP}] {str(PORT)}", listening_start_event
+        )
+
+        output_notifier.notify_output(
+            f"[{CLIENT_ALPHA_IP}] {str(PORT)} (?) open", sender_start_event
+        )
+
+        output_notifier.notify_output(
+            f"connect to [{CLIENT_ALPHA_IP}] from (UNKNOWN) [{CLIENT_BETA_IP}]",
+            connected_event,
+        )
+
+        output_notifier.notify_output("LAST_ACK", last_ack_event)
+        output_notifier.notify_output("TIME_WAIT", time_wait_event)
+
+        # registering on_stdout callback on both streams, cuz most of the stdout goes to stderr somehow
```

Comment:
I would move it as close as possible to the line that the comment refers to, to avoid confusion. In this case I think this comment should be before line 642.

Reply:
fixed